scripts/officer-parser.mjs

```diff
@@ -0,0 +1,98 @@
+import * as fs from 'fs';
+import { google } from "googleapis";
+import * as dotenv from "dotenv";
+
+dotenv.config();
+const SERVICE_ACCOUNT = process.env["SERVICE_ACCOUNT"];
+const DIRECTORY_SPREADSHEET_ID = process.env["DIRECTORY_SPREADSHEET_ID"];
+
+
+// await getGoogleSheetData('Officers!A2:K');
+writeToOutput(await getGoogleSheetData('Officers!A2:K'));
+
+
+////////////////////////////////////////////////////////
+// Helper Functions
+////////////////////////////////////////////////////////
+
+// Read data from Google sheets
+// using sheet range (eg: 'Week 1!A:H)
+async function getGoogleSheetData(range) {
+  const sheets = google.sheets({ version: 'v4' });
+
+  // Get JWT Token to access sheet
+  const service_account = JSON.parse(SERVICE_ACCOUNT);
+  const jwtClient = new google.auth.JWT(
+    service_account.client_email,
+    null, // or undefined, or an empty string (depends on your use case)
+    service_account.private_key,
+    ['https://www.googleapis.com/auth/spreadsheets'],
+  );
+
+  // Authorize the client
+  await jwtClient.authorize();
+
+  const committees = ["Board, Internal", "Board, External", "AI", "Cyber", "Design", "Studio", "Hack", "ICPC", "Teach LA", "W"];
+
+  // Get data from Google spreadsheets
+  try {
+    const res = await sheets.spreadsheets.values.get({
+      auth: jwtClient,
+      spreadsheetId: DIRECTORY_SPREADSHEET_ID,
+      range: range,
+    });
+
+    
+    let currCommittee = "President";
+    let offs = [];
+    const rows = res?.data.values;
+    for (let i = 0; i < rows.length; i++) {
+      const row = rows[i];
+      if (committees.includes(row[0])){
+        currCommittee = row[0];
+      } else if (row[1]) {
+        // Handle image URL
+        let image = row[10];
+        if (!image) {
+          image = 'https://t4.ftcdn.net/jpg/02/15/84/43/360_F_215844325_ttX9YiIIyeaR7Ne6EaLLjMAmy4GvPC69.jpg';  //try making this a reference under assests?
+        } else if (image.includes('https://drive.google.com')) {
```

Comment:
What you are probably looking for here is to check whether or not the URL starts with this string rather than if it includes `drive.google.com`. The following payload is a bypass that demonstrates the issue here and the potential fix.

```
> "https://www.evil.com?q=https://drive.google.com".includes("https://drive.google.com")
true
> "https://www.evil.com?q=https://drive.google.com".startsWith("https://drive.google.com")
false
> "https://drive.google.com/some-file".startsWith("https://drive.google.com")
true
```

Reply:
> What you are probably looking for here is to check whether or not the URL starts with this string rather than if it includes `drive.google.com`. The following payload is a bypass that demonstrates the issue here and the potential fix.
> 
> ```
> > "https://www.evil.com?q=https://drive.google.com".includes("https://drive.google.com")
> true
> > "https://www.evil.com?q=https://drive.google.com".startsWith("https://drive.google.com")
> false
> > "https://drive.google.com/some-file".startsWith("https://drive.google.com")
> true
> ```

I fail to see how this is a security issue with a bypass as the code segment just extracts a `fileid` from the url and turns it into a thumbnail image url.